### PR TITLE
ci: Exclude samtranslator.internal from interface scan

### PR DESCRIPTION
**Pull request by aahung**
### Description of changes

See #3026, an example of unnecessary workflow failure

### Checklist

- [ ] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [ ] Using correct values
    - [ ] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: bin/public_interface.py [hunk 1]
```diff
@@ -212,13 +215,19 @@ def main() -> None:
     subparsers = parser.add_subparsers(dest="command")
     extract = subparsers.add_parser("extract", help="Extract public interfaces")
     extract.add_argument("--module", help="The module to extract public interfaces", type=str, default="samtranslator")
+    extract.add_argument(
+        "--skipped-modules",
+        help="The modules separated by ',' that should be skipped",
```
Comment (acristin) [inline bin/public_interface.py:220]: Minor nit: `"The modules, separated by ',', that should be skipped"`

File: bin/public_interface.py [hunk 2]
```diff
@@ -212,13 +215,19 @@ def main() -> None:
...
+        default="samtranslator.internal",
+    )
     check = subparsers.add_parser("check", help="Check public interface changes")
     check.add_argument("original_json", help="The original public interface JSON file", type=Path)
     check.add_argument("new_json", help="The new public interface JSON file", type=Path)
     args = parser.parse_args()
 
     if args.command == "extract":
-        scanner = InterfaceScanner()
+        scanner = InterfaceScanner(skipped_modules=args.skipped_modules.split(","))
```
Comment (acristin) [inline bin/public_interface.py:230]: Should we trim the resulting strings in case the customer adds spaces between the comma and the next item?

**Review by acristin — APPROVED** [review]
Overall looks good; minor suggestion around UX that you're free to ignore or put off until later.

**Review by xazhao — APPROVED** [review]

File: bin/public_interface.py [hunk 3]
```diff
@@ -212,13 +215,19 @@ def main() -> None:
     subparsers = parser.add_subparsers(dest="command")
     extract = subparsers.add_parser("extract", help="Extract public interfaces")
     extract.add_argument("--module", help="The module to extract public interfaces", type=str, default="samtranslator")
+    extract.add_argument(
+        "--skipped-modules",
+        help="The modules separated by ',' that should be skipped",
+        type=str,
+        default="samtranslator.internal",
```
Comment (hoffa) [inline bin/public_interface.py:222]: You can use `nargs='+'` so you don't need the `split`.

**Review by aaythapa — APPROVED** [review]

**Review by hoffa — APPROVED** [review]

File: bin/public_interface.py (hunk 3 above)
Comment (aahung) [reply bin/public_interface.py]: nargs="*"

File: bin/public_interface.py (hunk 2 above)
Comment (aahung) [reply bin/public_interface.py]: no longer apply due to nargs

**Pull request merged**